presto-hive/src/test/java/com/facebook/presto/hive/TestHivePushdownFilterQueries.java

```diff
@@ -326,6 +334,28 @@ public void testDecimals()
         assertFilterProject("long_decimals[1] > 0.01", "count(*)");
     }
 
+    @Test
+    public void testSlice()
+    {
+        //char
+        assertFilterProject("orderkey = 8480", "char_array");
+        assertFilterProject("orderkey < 1000", "fixed_comment");
+
+        //varchar/char direct
+        assertFilterProject("comment is not NULL and linenumber=1 and orderkey<10", "comment");
+        assertFilterProject("comment is NULL", "count(*)");
+        assertFilterProject("length(comment) > 14 and orderkey < 150 and linenumber=2", "count(*)");
+        assertFilterProject("comment like '%fluf%'", "comment");
+        assertFilterProject("orderkey = 8480", "comment, fixed_comment");
+
+        //varchar/char dictionary
+        assertFilterProject("orderkey < 5000", "shipinstruct");
```

Comment:
- some test coverage appears to be missing, e.g. compactValues in dictionary reader, non-deterministic filters in both dictionary and direct; could you add more tests to ensure all the non-trivial code paths are covered?one way to improve coverage would be to add the following tests

```
assertFilterProject("shipmode in ('AIR', 'MAIL', 'RAIL')\n" +
                "AND shipinstruct in ('TAKE BACK RETURN', 'DELIVER IN PERSON')\n" +
                "AND substr(shipinstruct, 2, 1) = substr(shipmode, 2, 1)\n" +
                "AND shipmode = if(linenumber % 2 = 0, 'RAIL', 'MAIL')", "orderkey");

assertFilterProject("varchar_array[1] BETWEEN 'd' AND 'f'", "orderkey");
```